README_NEW.md

```diff
@@ -0,0 +1,349 @@
+# Go backend for MLflow
+
+In order to increase the performance of the tracking server and the various stores, we propose to rewrite the server and store implementation in Go.
+
+
+# Topics
+* [Installation](#installation)
+* [Mage](#mage)
+* [Development Setup](#development-setup)
+  * [Installation](#installation)
+  * [Run tests](#run-tests)
+  * [Tests state](#tests-state)
+* [Common information](#common-information)
+  * [MLflow source code](#mlflow-source-code)
+  * [Protos](#protos)
+  * [Request validation](#request-validation)
+  * [Data access](#data-access)
+  * [Linting](#linting)
+* [Usage](#usage)
+  * [Go Server](#go-server)
+  * [Building the Go binary](#building-the-go-binary)
+  * [Client-side Go implementation](#client-side-go-implementation)
+  * [Go store in Python](#go-store-in-python)
+* [Misc](#misc)
+  * [Debug Failing Tests](#debug-failing-tests)
+  * [Supported endpoints](#supported-endpoints)
+  * [Targeting Local Postgres in Python Tests](#targeting-local-postgres-in-python-tests)
+
+## Installation
+
+This package is not yet available on PyPI and currently requires the [Go SDK](https://go.dev) to be installed.
+
+You can then install the package via pip:
+```bash
+pip install git+https://github.com/jgiannuzzi/mlflow-go.git
+```
+
+## Mage
+
+This repository uses [mage](https://magefile.org/) to streamline some utility functions.
+
+```bash
+# Install mage (already done in the dev container)
+go install github.com/magefile/mage@v1.15.0
+
+# See all targets
+mage
+
+# Execute single target
+mage generate
+```
+
+The beauty of Mage is that we can use regular Go code for our scripting.  
+That being said, we are not married to this tool.
+
+## Development Setup
+
+### Installation
+
+```bash
```

Comment:
Consider rearranging the content to place `mage configure` at the top. Start with a brief explanation of its purpose, and then use an expandable section to provide more detailed information about its functionality.